File: src/main/java/Exercise_1.java
class Stack {
    static final int MAX = 1000;
    int top;
    int a[] = new int[MAX]; // Maximum size of Stack 

    boolean isEmpty() {
        //if stack has any elements return false or else true
        if (top >= 0) {
            return false;
        } else {
            return true;
        }
    }

    Stack() {
        //Initialize your constructor
        top = -1;
    }

    boolean push(int x) {
        //push to stack if top is less then size of the stack
        if (top < MAX - 1) {
            a[++top] = x;
            System.out.println("pushed element : " + x);
            return true;
        } else {
            System.out.println("stack limit exceeded");
            return false;
        }
    }

    int pop() {
        //If empty return 0 and print " Stack Underflow"
        //pop if there is one ore more elements in the stack
        if (top > -1) {
            int pop = a[top];
            top--;
            System.out.println("popped element : " + pop);
            return pop;
        } else {
            System.out.println("Stack Underflow");
            return 0;
        }

    }

    int peek() {
        //peek if stack has any values in it
        if (top > -1) {
            System.out.println("peeked element : " + a[top]);
            return a[top];
        } else {
            System.out.println("Stack Underflow");
            return 0;
        }
    }
}


// Driver code 
class Main {
    public static void main(String args[]) {
        Stack s = new Stack();
        s.push(10);
        s.push(20);
        s.push(30);
        System.out.println(s.pop() + " Popped from stack");
        s.peek();
    }
    //Time Complexity: O(1)
    //Space Complexity: O(n)
}

File: Exercise_1.ts
// O(1)
export class Stack {
    max = 1000;
    top: number = 0;
    elements = [];
    a = [this.max];

    constructor() { }

    push(x: number) {
        // Assign x at index top of array elements
        this.elements[this.top] = x;
        // Increment the top as previous index has a value assigned
        this.top = this.top + 1;
    }

    pop(): number {
        // Check if stack is empty, if yes return 0 or remove the last item
        if (this.elements.length === 0) {
            console.log('Stack Underflow');
            return 0;
        }
        
        return this.elements.pop();
    }

    peek(): number {
        // Return the top most element from the stack
        return this.elements[this.elements.length - 1];
    }

    isEmpty(): boolean {
        // Return if stack is empty
        return this.elements.length === 0;
    }

    count(n: number){
        if(n <= 0){
            return;
        }
        n = n-1;
        this.count(5);
    }
}

class Main {
    private stack: Stack = new Stack();

    constructor() {
        this.stack.push(10);
        this.stack.push(20);
        this.stack.push(30);
        this.stack.pop();
        this.stack.peek();
        this.stack.isEmpty();
        this.stack.count(5);
    }
}